src/Expander.hs

```diff
@@ -1074,3 +1075,36 @@ interpretMacroAction (MacroActionIdentEq how v1 v2) = do
 interpretMacroAction (MacroActionLog stx) = do
   liftIO $ prettyPrint stx >> putStrLn ""
   pure $ Right (ValueBool False) -- TODO unit type
+
+type NewScopeCheckTask = (Context, SplitCore)
+
+scopeCheckExpand :: ExpanderLocal -> SplitCore -> Expand ()
+scopeCheckExpand localState splitCore = do
+  phase <- currentPhase
+  let ctx = view expanderContext localState
+  root <-
+    case getRoot splitCore of
+      Nothing -> fail "internal error"
```

Comment:
We have an internal error constructor on the expressions. Can this be `throwError`?